Bottom: 0a66a67d3e66ac9a7d22c3f6f4444316bcbb05be
Top:    247be85436508a7bf5a3d00b53d97b8962bc6457
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-06-02 21:56:47 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index a122028..d89a968 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -230,6 +230,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
